lib/azure/armrest/armrest_service.rb

```diff
@@ -216,7 +216,9 @@ def rest_execute(options, http_method = :get, encode = true, max_retries = 3)
             if tries <= max_retries
               msg = "A rate limit or server side issue has occurred [#{err.http_code}]. Retry number #{tries}."
               Azure::Armrest::Configuration.log.try(:log, Logger::WARN, msg)
-              sleep_time = err.response.headers[:retry_after] || 30
+              sleep_time = (err.response.headers[:retry_after] || 30).to_i
+              sleep_time = 5 if sleep_time < 5     # 5 second minimum
+              sleep_time = 120 if sleep_time > 120 # 2 minute maximum
```

Comment:
Too bad we can't use the new #clamp method yet

https://blog.bigbinary.com/2016/12/13/ruby-2-4-adds-comparable-clamp-method.html

Reply:
If you're using more_core_extensions, we do have clamp, but it doesn't look like this gem uses more_core_extensions

https://github.com/ManageIQ/more_core_extensions/blob/master/lib/more_core_extensions/core_ext/numeric/clamp.rb

Reply:
@Fryguy didn't know about that, could be useful in the future!